### ISSUE TITLE
bump prometheus from 0.38.0 to 0.39.1  causes issues

### ISSUE THREAD

**Issue by codeboten** [open-telemetry/opentelemetry-collector-contrib#14882]
### Describe the issue you're reporting

Updating to the latest prometheus library causes the following issue, as per the log [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3228344565/jobs/5284317876):
```
	github.com/prometheus/prometheus/pkg/rulefmt: module github.com/prometheus/prometheus@latest found (v0.39.1), but does not contain package github.com/prometheus/prometheus/pkg/rulefmt
```

This needs to be addressed to allow the following PRs to move forward:
https://github.com/open-telemetry/opentelemetry-collector-contrib/pulls?q=is%3Apr+is%3Aclosed+prometheus+from+0.38.0+to+0.39.1+

**Comment by kevinschoonover** [comment]
This seems to also cause conflict between prometheusreceiver and lokiexporter
```
# otelcol-builder.yml
receivers:
  - import: go.opentelemetry.io/collector/receiver/otlpreceiver
    gomod: go.opentelemetry.io/collector v0.62.1
  - gomod: "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver v0.62.0"
exporters:
  - gomod: "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/lokiexporter v0.62.0"
```

```bash
>  mkdir -p /tmp/otel-test && builder --config=otelcol-builder-test.yaml --output-path=/tmp/otel-test
Flag --output-path has been deprecated, use config distribution::output_path
2022-10-13T23:50:40.823-0700	INFO	internal/command.go:121	OpenTelemetry Collector Builder	{"version": "dev", "date": "unknown"}
2022-10-13T23:50:40.823-0700	INFO	internal/command.go:154	Using config file	{"path": "otelcol-builder-test.yaml"}
2022-10-13T23:50:40.823-0700	INFO	builder/config.go:103	Using go	{"go-executable": "/run/current-system/sw/bin/go"}
2022-10-13T23:50:40.826-0700	INFO	builder/main.go:76	Sources created	{"path": "/tmp/otel-test"}
Error: failed to update go.mod: exit status 1. Output: "go: finding module for package github.com/prometheus/prometheus/model/metadata\ngo.opentelemetry.io/collector/cmd/builder imports\n\tgithub.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver imports\n\tgithub.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver/internal imports\n\tgithub.com/prometheus/prometheus/model/metadata: package github.com/prometheus/prometheus/model/metadata provided by github.com/prometheus/prometheus at latest version v0.39.1 but not at required version v1.8.2-0.20220303173753-edfe657b5405\n"
```

**Comment by disfluxly** [comment]
Any update on this by chance. You effectively can't build a collector using builder that has both the Loki Exporter & the Prometheus Exporter/Receiver in it at this time.

**Comment by mar4uk** [comment]
in version 0.67.0 this issue is fixed. Shouldn't reproduce anymore

```
dist:
  name: otelcol-dev
  description: "Basic OTel Collector distribution for Developers"
  output_path: ./otelcol-dev
receivers:
  - import: go.opentelemetry.io/collector/receiver/otlpreceiver
    gomod: go.opentelemetry.io/collector v0.67.0
  - gomod: "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver v0.67.0"
exporters:
  - gomod: "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/lokiexporter v0.67.0"
```

```
./ocb --config manifest.yaml
2022-12-12T10:11:55.196Z	INFO	internal/command.go:125	OpenTelemetry Collector Builder	{"version": "0.67.0", "date": "2022-12-09T00:36:07Z"}
2022-12-12T10:11:55.197Z	INFO	internal/command.go:158	Using config file	{"path": "manifest.yaml"}
2022-12-12T10:11:55.197Z	INFO	builder/config.go:107	Using go	{"go-executable": "/usr/local/go/bin/go"}
2022-12-12T10:11:55.198Z	INFO	builder/main.go:76	Sources created	{"path": "./otelcol-dev"}
2022-12-12T10:11:55.525Z	INFO	builder/main.go:118	Getting go modules
2022-12-12T10:11:55.578Z	INFO	builder/main.go:87	Compiling
2022-12-12T10:12:23.614Z	INFO	builder/main.go:99	Compiled	{"binary": "./otelcol-dev/otelcol-dev"}
```

**Comment by mar4uk** [comment]
@codeboten can we close this issue?
The problem was solved in the v0.67.0 version of the collector contib.
Currently, github.com/prometheus/prometheus v0.41.0 is used
